src/participant.rs

```diff
@@ -355,4 +369,30 @@ mod tests {
         let frost_id2 = id.to_frost_identifier();
         assert_eq!(frost_id1, frost_id2);
     }
+
+    #[test]
+    fn test_authenticated_data() {
+        let secret = Secret::random(thread_rng());
+        let id = secret.to_identity();
+
+        let data = b"hello world";
+        let signature = secret.sign(data);
+
+        id.verify_data(data, &signature.to_bytes())
+            .expect("verification failed");
+    }
+
+    #[test]
+    fn test_authenticated_data_bad_signature() {
```

Comment:
Nit: for completeness we should also have a test for verifying invalid data

Reply:
Added